specification/ml/put_job/MlPutJobRequest.ts

```diff
@@ -85,6 +89,10 @@ export interface Request extends RequestBase {
      * A list of job groups. A job can belong to no groups or many.
      */
     groups?: string[]
+    /**
+     * The identifier for the anomaly detection job. This identifier can contain lowercase alphanumeric characters (a-z and 0-9), hyphens, and underscores. It must start and end with alphanumeric characters.
+     */
+    job_id?: Id
```

Comment:
I'm not sure why we'd support putting a job_id in the body when it's required in the path. I think I vaguely remember knowing that if you *did* supply it in the body you got an error if it didn't match what was in the path, so I'm not sure why anyone would bother adding it. This is likely another question for someone like @droberts195

Reply:
I just did a quick test and this is the error I get if the job_id in the body differs from the job_id in the path:
> {
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "Inconsistent job_id; 'total-requests3' specified in the body differs from 'total-requests2' specified as a URL argument"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "Inconsistent job_id; 'total-requests3' specified in the body differs from 'total-requests2' specified as a URL argument"
  },
  "status" : 400
}

Therefore unless someone else knows why you'd ever want to use the value in the body, I'd recommend putting something like this in the description: "If specified, this value must match the path parameter; otherwise, an error occurs."

Reply:
Allowing the job ID in the body was just to make it easier for people who want to store configs as a single JSON document in some other location to then add into multiple clusters. That JSON document can then include the job ID. It has to be copied into the URL to add the config, but doesn’t also need to be removed from the JSON document.

Reply:
Thanks for the info!  I've added a suggestion for the description.